homework/index.js

```diff
@@ -29,26 +33,149 @@
     return elem;
   }
 
-  function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+  function renderContributorsDetails(contributor, contributorsList) {
+    const contributorItem = createAndAppend('li', contributorsList);
+    const contributorDetails = createAndAppend('div', contributorItem, {
+      class: 'contributor',
+    });
+
+    createAndAppend('a', contributorDetails, {
+      href: contributor.html_url,
+      target: `_blank`,
+      text: contributor.login,
+    });
+
+    createAndAppend('img', contributorDetails, {
+      src: contributor.avatar_url,
+      alt: 'avatar',
+    });
+
+    createAndAppend('span', contributorDetails, {
+      text: contributor.contributions,
+    });
+  }
+
+  function buildTableRow(title, parent, content) {
+    const row = createAndAppend(`tr`, parent);
+    createAndAppend(`th`, row, { text: `${title}:` });
+    createAndAppend(`td`, row, { text: `${content}` });
+    return row;
+  }
+
+  function renderRepoDetails(repo, mainSection) {
+    const repoSection = createAndAppend('section', mainSection, {
+      class: 'repo-container',
+    });
+    const repoDetails = createAndAppend('div', repoSection, {
+      class: 'details',
+    });
+    const table = createAndAppend(`table`, repoDetails);
+
+    const firstRow = buildTableRow(`Repository`, table, ``);
+    createAndAppend(`a`, firstRow.lastChild, {
+      href: repo.html_url,
+      target: `_blank`,
+      text: repo.name,
+    });
+
+    const descriptionRow = buildTableRow(`Description`, table, ``);
+    const description = repo.description
+      ? { text: repo.description }
+      : { text: 'No Description', class: 'alert-no-description' };
+    createAndAppend(`Description`, descriptionRow.lastChild, description);
+
+    buildTableRow(`Forks`, table, repo.forks);
+
+    const date = new Date(repo.updated_at);
+    buildTableRow(`Updated`, table, date.toLocaleString());
+  }
+
+  function selectRepo(repos, select, root) {
+    const mainSection = createAndAppend('main', root, {
+      class: 'main-container',
+    });
+
+    select.addEventListener('change', () => {
+      mainSection.innerHTML = '';
+
+      const repo = repos[select.value];
+      renderRepoDetails(repo, mainSection);
+
+      const contributorsSection = createAndAppend('section', mainSection, {
+        class: 'contributors-container',
+      });
+      createAndAppend('h3', contributorsSection, { text: 'Contributions' });
+      const contributorsList = createAndAppend('ul', contributorsSection);
+
+      const CONTRIBUTORS_URL = `${HYF_BASE_URL}/repos/HackYourFuture/${repo.name}/contributors`;
```

Comment:
You should not construct this url yourself. Use the `contributors_url` from the repo object.

Reply:
I'll try